static/app/components/events/contextSummary/index.tsx

```diff
@@ -35,85 +32,93 @@ const KNOWN_CONTEXTS: Context[] = [
     unknownTitle: t('Unknown Runtime'),
     omitUnknownVersion: true,
   },
-  {keys: ['client_os', 'os'], Component: ContextSummaryOS},
-  {keys: ['device'], Component: ContextSummaryDevice},
-  {keys: ['gpu'], Component: ContextSummaryGPU},
+  {
+    keys: ['client_os', 'os'],
+    Component: ContextSummaryOS,
+  },
+  {
+    keys: ['device'],
+    Component: ContextSummaryDevice,
+  },
+  {
+    keys: ['gpu'],
+    Component: ContextSummaryGPU,
+  },
 ];
 
 type Props = {
   event: Event;
 };
 
 function ContextSummary({event}: Props) {
-  let contextCount = 0;
+  const filteredContexts = KNOWN_CONTEXTS.filter(makeContextFilter(event));
 
-  // Add defined contexts in the declared order, until we reach the limit
-  // defined by MAX_CONTEXTS.
-  let contexts = KNOWN_CONTEXTS.filter(context => filterContexts(event, context)).map(
-    ({keys, Component, unknownTitle, omitUnknownVersion}) => {
-      if (contextCount >= MAX_CONTEXTS) {
-        return null;
-      }
+  // XXX: We want to have *at least* MIN_CONTEXTS, so we first find all the
+  // contexts that have data, if that doesn't complete our MIN_CONTEXTS, we add
+  // in contextrs that have no data.
```

Comment:
I rewrote this to be a lot more clear, it's explained here what it's trying to do.